### ISSUE TITLE
xmrig proxy 0 hash glitch

### ISSUE THREAD

**Issue by jamiekarvans** [UnamSanctam/SilentCryptoMiner#2333]
Hi unam, i have setup zephyr on herominers and i'm using Xmrig Proxy but as you can see in the pictures sometimes the hash drops to 0 while i can check in the unam panel everything is working fine and miners are connected and are online.
what could be the issue?
![Screenshot_20240413_005115_Microsoft Remote Desktop](https://github.com/UnamSanctam/SilentCryptoMiner/assets/69362068/cb92e442-d264-46d2-ad7f-e3cac5ffeee4)

**Comment by UnamSanctam** [comment]
xmrig-proxy calculates the hashrate in the same manner that pools do it, by the amount of shares found over some interval. As you can see the hashrate says 0 when the amount of "shares" is the same (for example 56063) for some time. You can make it "report" the hashrate more often by enabling "fake" shares using the custom diff option, for example set `"custom-diff-stats"` to `true` and then set `"custom-diff"` to some difficulty, for example `10000` which will send a share to your xmrig-proxy more often (but it's not sent to the pool unless it has reached the pools share difficulty).

**Comment by jamiekarvans** [comment]
So it's literally performance useless, as it only shows me hashrate and
doesn't change anything about actual hashrate on the mining pool?

**Comment by UnamSanctam** [comment]
Yes the custom diff is purely for statistical reasons, you can also press 'H' in your xmrig-proxy to get the average hashrate over time which would be more accurate. When only counting in 1 minute intervals then it showing 0 H/s is normal with your amount of miners and the pool share difficulty.
